src/kv/LevelDBStore.cc

```diff
@@ -270,11 +270,13 @@ int LevelDBStore::get(const string &prefix,
   assert(value && (value->length() == 0));
   utime_t start = ceph_clock_now(g_ceph_context);
   int r = 0;
-  KeyValueDB::Iterator it = get_iterator(prefix);
-  it->lower_bound(key);
-  if (it->valid() && it->key() == key) {
-    value->append(it->value_as_ptr());
-  } else {
+  leveldb::ReadOptions options;
+  std::string value;
```

Comment:
Thank you, Kefu!

On Wed, Jun 1, 2016 at 7:14 PM, Kefu Chai <notifications@github.com> wrote:

> In src/kv/LevelDBStore.cc
> <https://github.com/ceph/ceph/pull/9215#discussion_r65474473>:
>
> > @@ -270,11 +270,13 @@ int LevelDBStore::get(const string &prefix,
> >    assert(value && (value->length() == 0));
> >    utime_t start = ceph_clock_now(g_ceph_context);
> >    int r = 0;
> > -  KeyValueDB::Iterator it = get_iterator(prefix);
> > -  it->lower_bound(key);
> > -  if (it->valid() && it->key() == key) {
> > -    value->append(it->value_as_ptr());
> > -  } else {
> > +  leveldb::ReadOptions options;
> > +  std::string value;
>
> @jjhuo <https://github.com/jjhuo>
>
> git remote add markhpc https://github.com/markhpc/ceph.git
> git fetch markhpc
> git checkout --track markhpc/wip-9215
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ceph/ceph/pull/9215/files/8afef6d56313c30ab60d62a134ebdd3857b4bca8#r65474473>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGfCpV0crB2iNL1btANvATPl8WFRH8MOks5qHjyRgaJpZM4Iiuwc>
> .
>